nixos/modules/services/continuous-integration/gitlab-runner.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.gitlab-runner;
+  configFile = pkgs.writeText "config.toml" cfg.configText;
+in
+{
+  options.services.gitlab-runner = {
+    enable = mkEnableOption "Gitlab Runner";
+
+    configText = mkOption {
+      description = "Verbatim config.toml to use";
```

Comment:
yes, but it must be interpolated to strings manually instead of using data structures,
which is more error prone. (using `remarshal` will always produce valid toml)

This is an example:
https://github.com/Mic92/nixpkgs/blob/8fddcad3f9831d1961ffd8f6f35f14c759a99cdd/nixos/modules/services/monitoring/telegraf.nix#L34